packages/mobile/src/send/utils.ts

```diff
@@ -171,21 +180,44 @@ export function showLimitReachedError(
   return showError(ErrorMessages.PAYMENT_LIMIT_REACHED, ALERT_BANNER_DURATION, translationParams)
 }
 
+// exported for testing
+export function* _updateRecipientCache(recipient: RecipientWithQrCode) {
+  if (!recipient.e164PhoneNumber) {
+    return
+  }
+  const addressToE164Number: AddressToE164NumberType = yield select(addressToE164NumberSelector)
+  const recipientCache: NumberToRecipient = yield select(recipientCacheSelector)
+
+  recipientCache[recipient.e164PhoneNumber] = {
+    contactId: '',
+    ...recipient,
+    kind: RecipientKind.Contact,
+  }
+  addressToE164Number[recipient.address] = recipient.e164PhoneNumber
```

Comment:
This makes me nervous because it overwrites any existing mapping without having any type of check for security or correctness. I don't know the best way to mitigate this concern atm.

I am imagining a scenario where someone creates an account with a popular but unverified phone number. Anyone that scan's that QR, will have that phone number overwritten with the QR owner's address even though the phone number is not verified

Reply:
Yeah, I agree, I was a bit lazy tbh, I found this code in some old PR and took it from there. I don't think we should be using phone numbers from this without checking if they're verified, how about this:

Instead of reusing these caches that use phone numbers, create a new one that maps `address -> display name` that's used as fallback if none of the others have info about the transfer. This way the display name in the QR would only be used if the alternative is to show 'Unknown' in the feed.

Reply:
Did the changes I suggested above, let me know what you think :)

Reply:
I like it! I think this is a good short-term solution until we get CIP8 integrated. Please remind whoever implements the CIP8 work to remove this cache.